apple-nav/src/components/NavBar.js

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import { Route } from 'react-router-dom';
+import styled from 'styled-components'
+import PropTypes from 'prop-types';
+import NavLinks from './NavLinks';
+import SubNav from './SubNav';
+
+const Header = styled.header `
+background: black;
+opacity: .80;
+display: flex;
+flex-flow: row nowrap;
+justify-content: space-around;
+align-items: center;
+`
+
+const NavBar = (props) => {
+return (
+    <React.Fragment>
+    <Header>
+        <NavLinks list={props.list} handleClick={props.handleClick} />
+    </Header>
+    
+    <Route
+    path={'/:category'} 
+    render={({match})=> 
+        <SubNav {...props} list={props.list} match={match}/>}
```

Comment:
The `{...props}` includes `match`, `location`, and `history`. You use those a lot in React-Router so anytime you see {...props}, it's passing those 3. So you don't really need to pass match down. (although the way you did your assignment, you still need to pass it down, because just changing line 26 `match` to `props` breaks the whole thing.)